### ISSUE TITLE
Markdown formatting doesn't handle URL with underscore

### ISSUE THREAD

**Issue by Eronarn** [mattermost/platform#1198]
I wrote a Buildbot extension (in Python) to post status updates directly to Mattermost using the API. This generates Markdown with a template like this:

```
        template = "#### **{result}** on branch '{branch}' in builder '{builder}'{result_details}\n" \
                   "*build reason: {reason}*\n" \
                   "*build details: [{link}]({link})*"
```

In Mattermost, the underscore is parsed as the end of the URL: ![screen shot 2015-10-27 at 10 09 08 am](https://cloud.githubusercontent.com/assets/536435/10760485/2a9cb15e-7c93-11e5-8da2-2c8301f4d822.png)

But if I do this on github, the same URL is highlighted in full: [nrg-buildmaster.rc.fas.harvard.edu/lib/builders/pylib_deploy/builds/5](nrg-buildmaster.rc.fas.harvard.edu/lib/builders/pylib_deploy/builds/5)

**Comment by it33** [comment]
Thanks @Eronarn! Definitely need to fix this, 

Very cool to see your buildbot running!

**Comment by hmhealey** [comment]
This has been fixed by #1218
